tidy3d/components/simulation.py

```diff
@@ -3570,6 +3580,33 @@ def _validate_time_monitors_num_steps(self) -> None:
                     "at which the monitor stores data."
                 )
 
+    def _validate_monitor_simulation_frequency_range(self) -> None:
+        """Rise an error if any DFT monitors have frequencies outside of the simulation frequency range."""
+        source_ranges = [source.source_time.frequency_range() for source in self.sources]
+        if not source_ranges:
+            return
+
+        freq_range_min = (
+            min((freq_range[0] for freq_range in source_ranges), default=0.0)
+            * MIN_MONITOR_FREQUENCY_RANGE_PARAMETER
+        )
+        freq_range_max = (
+            max((freq_range[1] for freq_range in source_ranges), default=0.0)
+            * MAX_MONITOR_FREQUENCY_RANGE_PARAMETER
+        )
+
+        for monitor in self.monitors:
+            if not isinstance(monitor, FreqMonitor):
+                continue
+
+            freqs = np.array(monitor.freqs)
+            if freqs.min() < freq_range_min or freqs.max() > freq_range_max:
+                raise SetupError(
+                    f"Time monitor '{monitor.name}' contains frequencies "
+                    f"outside of the simulation frequency range ({freq_range_min:2e}, {freq_range_max:2e})"
```

Comment:
We should also specify the margin, i.e. the `MAX_MONITOR_FREQUENCY_RANGE_PARAMETER`.